modelon/impact/client/experiment_definition.py

```diff
@@ -226,14 +317,174 @@ def to_dict(self):
+        self.variable_modifiers = {}
+
+    def with_modifiers(self, modifiers=None, **modifiers_kwargs):
+        """Sets the modifiers variables for an experiment extension.
+
+        Parameters:
+
+            modifiers --
+                A dictionary of variable modifiers. Could be used if
+                modifiers keys conflict with python identifiers or keywords.
+                Default: None.
+
+            modifiers_kwargs --
+                A keyworded, variable-length argument list of variable
+                modifiers.
+
+        Example::
+
+            fmu = model.compile().wait()
+            simulation_options = custom_function.get_simulation_options()
+            .with_values(ncp=500)
+            solver_options = {'atol':1e-8}
+            simulate_ext = SimpleExperimentExtension().with_modifiers({'PI.k': 40})
+            simulate_ext = SimpleExperimentExtension(
+            {'start_time': 0.0, 'final_time': 4.0},
+            solver_options,
+            simulation_options
+            ).with_modifiers({'PI.k': 40})
+
+        """
+        modifiers = {} if modifiers is None else modifiers
+        modifiers_aggregate = {**modifiers, **modifiers_kwargs}
+        new = SimpleExperimentExtension(
+            self.parameter_modifiers,
+            self._solver_options,
+            self._simulation_options,
+            self.simulation_log_level,
+        )
+
+        for variable, value in modifiers_aggregate.items():
+            if isinstance(value, Operator):
+                raise ValueError(
+                    "Range operator is not supported when using extentions"
+                    " in the experiment"
+                )
+            new.variable_modifiers[variable] = value
+        return new
+
+    def to_dict(self):
+        """Returns the experiment extensions as a dictionary.
+
+        Returns:
+
+            extensions_dict --
+                A dictionary containing the experiment extensions.
+
+        Example::
+
+            fmu = model.compile().wait()
+            simulation_options = custom_function.get_simulation_options()
+                .with_values(ncp=500)
+            solver_options = {'atol':1e-8}
+            simulate_ext = SimpleExperimentExtension(
+            {'start_time': 0.0, 'final_time': 4.0},
+            solver_options,
+            simulation_options,
+            ).with_modifiers({'PI.k': 40})
+            simulate_ext.to_dict()
+        """
+        return {
+            "extensions": [
+                {
+                    "modifiers": {"variables": self.variable_modifiers},
+                    "analysis": {
+                        "parameters": self.parameter_modifiers,
+                        "simulationOptions": self._simulation_options,
+                        "solverOptions": self._solver_options,
+                        "simulationLogLevel": self.simulation_log_level,
+                    },
+                }
+            ]
+        }
```

Comment:
As this is `to_dict` for a single extensions, it makes sense if it only returns `{'modifiers': ...., 'analysis': ...}` which represents a single extension.